File: connect.py
import socket
import ssl
import sys
import pprint

context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
context.verify_mode = ssl.CERT_REQUIRED
context.check_hostname = True
context.load_default_certs()
context.load_verify_locations(cafile='ca.crt')

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
ssl_sock = context.wrap_socket(s, server_hostname="tls_drama")
ssl_sock.connect((sys.argv[-1], 2103))
cert = ssl_sock.getpeercert()
pprint.pprint(cert)
ssl_sock.sendall(b'Hello')
print(ssl_sock.recv(1024))

File: connect.rb
require 'socket'
require 'openssl'

host = ARGV[-1]
port = 2103

socket = TCPSocket.open(host,port)
ssl_context = OpenSSL::SSL::SSLContext.new(:TLSv1)
ssl_context.cert = OpenSSL::X509::Certificate.new(File.open("server.crt"))
ssl_context.key = OpenSSL::PKey::RSA.new(File.open("server.key"))
ssl_context.set_params(verify_mode: OpenSSL::SSL::VERIFY_PEER)
ssl_socket = OpenSSL::SSL::SSLSocket.new(socket, ssl_context)
ssl_socket.sync_close = true
ssl_socket.connect

ssl_socket.puts "Hello"
print ssl_context.gets

ssl_socket.close
